Modules/_datetimemodule.c

```diff
@@ -3711,6 +3735,26 @@ time_str(PyDateTime_Time *self)
     return _PyObject_CallMethodId((PyObject *)self, &PyId_isoformat, NULL);
 }
 
+/* Return new time from time string, using _strptime._parse_isotime(). */
+static PyObject *
+time_fromisoformat(PyObject *cls, PyObject *args)
+{
+    static PyObject *module = NULL;
+    PyObject *string;
+
+    if (!PyArg_ParseTuple(args, "U:fromisoformat", &string))
+        return NULL;
+
+
```

Comment:
You've used only one blank line in the other places (which is IMHO more beautiful).

Also, do you reckon it would make sense to have a single function like:

```

static PyObject *
strptime_fromisoformat(PyObject *cls, PyObject *args, const char *method_name)
{
    static PyObject *module = NULL;
    PyObject *string;

    if (!PyArg_ParseTuple(args, "U:fromisoformat", &string))
        return NULL;

    if (module == NULL) {
        module = PyImport_ImportModule("_strptime");
        if (module == NULL)
            return NULL;
    }

    return PyObject_CallMethod(module, method_name, "OO", cls, string);
}

static PyObject *
date_fromisoformat(PyObject *cls, PyObject *args)
{
    return strptime_fromisoformat(cls, args, "_parse_isodate");
}

static PyObject *
time_fromisoformat(PyObject *cls, PyObject *args)
{
    return strptime_fromisoformat(cls, args, "_parse_isotime");
}

static PyObject *
datetime_fromisoformat(PyObject *cls, PyObject *args)
{
    return strptime_fromisoformat(cls, args, "_parse_isodatetime");
}










```